prox_reader.py

```diff
@@ -0,0 +1,55 @@
+""" 
+A class that abstracts the proximity sensor hardware interface and allows for the simultaneous return of front and back sensor measurements for the PowerPlant project
+"""
+
+from sensor_switch import MuxSwitcher
+from prox_interface import ProxInterface
+
+FRONT = 1
+BACK = 2
+
+
+class ProxReader(object):
+    """
+    Implementation of ProxReader functionality which includes taking a measurement from both the front and back sensors at the same time, and the ability to select which units should be returned.
+    """
+    
+    def __init__(self, units):
+        """
+        Initializes the measurement with the desired units to return.
+        Units options:
+            'm'
+            'cm'
+            'mm' (default)
+        """
+        
+        self.units = units
+        
+    def measure():
+        """
+        Takes measurement from front sensor and then switches to the back sensor. Though the measurements are not taken at the exact same time, the measurements should be close enough in time to serve the purpose of providing a distance measurement for the Power Plant project.
+        """
+        
+        bus = MuxSwitcher()
+        bus.sensor_select(FRONT)
+        prox = ProxInterface()
+        front = prox.read_prox()
+        
+        bus = MuxSwitcher()
+        bus.sensor_select(BACK)
+        prox = ProxInterface()
```

Comment:
I'm not certain whether a similar thing could be done with the prox variable since we found the prox driver is doing a lot of stuff behind the scenes. Do you think it'd work?

Reply:
I'll give it a try the next time I get a chance! Good suggestion.

Reply:
It seemed to work so I added it